cache/layerfactories.go

```diff
@@ -0,0 +1,30 @@
+package cache
+
+func MakeMemoryReadWriterFactory(minLayer uint) LayerFactory {
+	return func(layer uint) LayerReadWriter {
+		if layer >= minLayer {
+			return &SliceReadWriter{}
+		}
+		return nil
+	}
+}
+
+func MakeMemoryReadWriterFactoryForLayers(layers []uint) LayerFactory {
```

Comment:
I considered this and decided against it because that would make using it more cumbersome (constructing a map is harder than a slice) and this code isn't called very often, so performance isn't as much of an issue.